### PR TITLE
[#29] As a user, I see the list of comments of an article

### LINKED ISSUE #29: As a user, I see the list of comments of an article
## Why
The user can view a list of existing comments on any article in the `Comments History` screen.

## Acceptance Criteria
- [ ] Add the top navigation bar and the `Back` button by reusing the UI layout from #19.
- [ ] Update the navigation bar title with text: `Comments History`.
- [ ] Add a vertical table view for containing the list of comments below the top navigation bar.
- [ ] Create a comment table view cell with UI layout: A comment body label, a commenter avatar image view,  a commenter name label, A comment `updatedAt` date label as shown in the below design.
- [ ] The comment table view cell's height should be dynamic based on the comment's content (body).

## Resources

- Sample screen UI layout:
Screen Shot 2021-08-21 at 16 51 08 copy

- Sample table view cell UI layout:
Screen Shot 2021-08-21 at 17 25 14

### PR DESCRIPTION
**Pull request by markgravity**
Resolved #29 

## What happened

Create comments screen UI

## Insight

- [x] Add the top navigation bar and the `Back` button by reusing the UI layout from #19.
- [x] Update the navigation bar title with text: `Comments History`.
- [x] Add a vertical table view for containing the list of comments below the top navigation bar.
- [x] Create a comment table view cell with UI layout: A comment body label, a commenter avatar image view,  a commenter name label, A comment `updatedAt` date label as shown in the below design.
- [x] The comment table view cell's height should be dynamic based on the comment's content (body).

## Proof Of Work

Screen Shot 2021-09-15 at 14 42 12

### CODE REVIEW

File: NimbleMedium/Sources/Presentation/Modules/FeedComments/FeedCommentRow/FeedCommentRow.swift [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                    .placeholder { defaultAvatar }
+                    .resizable()
+                    .frame(width: 25.0, height: 25.0)
+            } else {
+                defaultAvatar
+            }
+            Text("cypress")
+                .foregroundColor(.green)
+            Text("August 12, 2021")
+                .foregroundColor(.gray)
```
Comment (minhnimble) [inline NimbleMedium/Sources/Presentation/Modules/FeedComments/FeedCommentRow/FeedCommentRow.swift:47]: please add TODO for hardcoded values 🙏

File: NimbleMedium/Sources/Presentation/Modules/FeedComments/FeedCommentsView.swift [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+private extension FeedCommentsView {
+
+    struct Content: View {
+
+        let view: FeedCommentsView
+
+        var body: some View {
+            ScrollView(.vertical) {
+                LazyVStack(alignment: .leading, spacing: 12.0) {
+                    ForEach(1...3, id: \.self) { _ in
```
Comment (minhnimble) [inline NimbleMedium/Sources/Presentation/Modules/FeedComments/FeedCommentsView.swift:27]: please add TODO for hardcoded values 🙏

File: NimbleMedium/Sources/Presentation/Modules/FeedComments/FeedCommentRow/FeedCommentRow.swift [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            if let url = try? "https://cdn3.iconfinder.com/data/icons/avatars-15/64/_Ninja-2-512.png".asURL() {
+                // FIXME: It blocks UI
+                WebImage(url: url)
+                    .placeholder { defaultAvatar }
+                    .resizable()
+                    .frame(width: 25.0, height: 25.0)
+            } else {
+                defaultAvatar
+            }
+            Text("cypress")
```
Comment (minhnimble) [inline NimbleMedium/Sources/Presentation/Modules/FeedComments/FeedCommentRow/FeedCommentRow.swift:44]: Just curious how do we handle long name here, is it going to go exceed the screen? 🤔

Comment (markgravity) [reply NimbleMedium/Sources/Presentation/Modules/FeedComments/FeedCommentRow/FeedCommentRow.swift:44]: @minhnimble  Still looks decent 😁 
Screen Shot 2021-09-20 at 16 45 19

**Review by suho — APPROVED** [review]

**Review by minhnimble — APPROVED** [review]
LGTM 🚀

File: NimbleMedium/Sources/Presentation/Modules/FeedComments/FeedCommentRow/FeedCommentRow.swift [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+//  Created by Mark G on 15/09/2021.
+//
+
+import SwiftUI
+import SDWebImageSwiftUI
+
+struct FeedCommentRow: View {
+
+    var body: some View {
+
```
Comment (minhnimble) [inline NimbleMedium/Sources/Presentation/Modules/FeedComments/FeedCommentRow/FeedCommentRow.swift:14]: can remove this line break 🙏

File: NimbleMedium/Sources/Presentation/Modules/FeedComments/FeedCommentsView.swift [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+//
+//  Created by Mark G on 15/09/2021.
+//
+
+import SwiftUI
+
+struct FeedCommentsView: View {
+
+    var body: some View {
+        Content(view: self)
```
Comment (minhnimble) [inline NimbleMedium/Sources/Presentation/Modules/FeedComments/FeedCommentsView.swift:13]: Please bring the sub-views from the `Content` struct into this var `body` directly, thanks 🙏

**Review by minhnimble — CHANGES_REQUESTED** [review]

**Review by minhnimble — APPROVED** [review]
LGTM 🚀

**Pull request merged**